### PR TITLE
Add Flex Dapps proposal implementation

### PR DESCRIPTION
**Pull request by sporejack**
# 10 - Flex Dapps
#10 

# Things to Check
 - [ ] USDC call uses 6 decimals
 - [ ] TCR call uses 18 decimals
 - [ ] Flex Dapps Ethereum address matches [their ENS registration](https://app.ens.domains/name/flexdapps.eth)
 - [ ] Timestamp is correct (against Snapshot -- see issue for link)

### CODE REVIEW

File: proposals/10-flex-dapps-proposal.md [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+
+## Issue Link
+[#10](https://github.com/tracer-protocol/proposals/issues/10)
+
+## Implementation Discussion
+N/A
+
+## Implementation
+Targets: [`0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48`,`0x707b6be09028e78d2a667db7596b2803c112f9b2`]
+Data: [`0xa9059cbb000000000000000000000000dd140cb2847f6186a9df3b25ae1e135052b0932100000000000000000000000000000000000000000000000000000006fc23ac00`,`0x0d535883000000000000000000000000dd140cb2847f6186a9df3b25ae1e135052b0932100000000000000000000000000000000000000000001a784379d99db420000000000000000000000000000009c4a4204b79dd291d6b6571c5be8bbcd0622f05000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009c0000000000000000000000000000000000000000000000000000000060a5c2a0`]
```
Comment (ScaredIbis) [inline proposals/10-flex-dapps-proposal.md:12]: Would have thought this should match the final output

**Review by ScaredIbis — CHANGES_REQUESTED** [review]
Contract calls all look correct,  just had some minor feedback on the documentation. Verified the following:

 - [x] USDC call uses 6 decimals
 - [x] USDC address is correct
 - [x] TCR call uses 18 decimals
 - [x] TCR address is correct
 - [x] TCR vesting contract address is correct
 - [x] Flex Dapps Ethereum address matches [their ENS registration](https://app.ens.domains/name/flexdapps.eth)
 - [x] Timestamp is correct as per [snapshot](https://vote.tracer.finance/#/tracer.eth/proposal/QmdTqbvQ3gBDbMoaWNP6WmESGurFdkU4fcnRoNrQadNcAc)

File: proposals/10-flex-dapps-proposal.md [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+                },
+                {
+                    "name": "vestingWeeks",
+                    "type": "uint256",
+                    "value": "156"
+                },
+                {
+                    "name": "startTime",
+                    "type": "uint256",
+                    "value": "1621476000"
```
Comment (ScaredIbis) [inline proposals/10-flex-dapps-proposal.md:90]: on second thought this timestamp corresponds to Thursday, 20 May 2021 12:00:00 GMT+10 but not sure what timezone the [snapshot](https://vote.tracer.finance/#/tracer.eth/proposal/QmdTqbvQ3gBDbMoaWNP6WmESGurFdkU4fcnRoNrQadNcAc) would be displaying times in, wouldn't be suprised if that was UTC

File: proposals/10-flex-dapps-proposal.md [hunk 3]
```diff
@@ -0,0 +1,197 @@
...
+TARGET:
+    0x9C4A4204B79dd291D6b6571C5BE8BbcD0622F050 (https://etherscan.io/address/0x9C4A4204B79dd291D6b6571C5BE8BbcD0622F050)
+DATA:
+    0xa9059cbb000000000000000000000000707b6be09028e78d2a667db7596b2803c112f9b200000000000000000000000000000000000000000001a784379d99db42000000
+
+============ Call 2 ============
+TARGET:
+    0x707b6be09028e78d2a667db7596b2803c112f9b2 (https://etherscan.io/address/0x707b6be09028e78d2a667db7596b2803c112f9b2)
+DATA:
+    0x0d535883000000000000000000000000dd140cb2847f6186a9df3b25ae1e135052b0932100000000000000000000000000000000000000000001a784379d99db420000000000000000000000000000009c4a4204b79dd291d6b6571c5be8bbcd0622f05000000000000000000000000000000000000000000000000000000000000000010000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009c0000000000000000000000000000000000000000000000000000000060a5c2a0
```
Comment (raymogg) [inline proposals/10-flex-dapps-proposal.md:117]: Looks like the payload here has `isFixed` equal to `true` (note the 1 between the token address and cliff weeks

File: proposals/10-flex-dapps-proposal.md (hunk 2 above)
Comment (antoniopgs) [reply proposals/10-flex-dapps-proposal.md:90]: @ScaredIbis I already looked into that. Snapshot converts dates to your local timezone.
Explanation here: https://github.com/tracer-protocol/proposals/issues/16

Leaving this here for anyone else who needs to calculate block.timestamps of snapshot dates:
```javascript
/*
    Snapshot converts dates to your system's current timezone, but Date constructor converts from system back to UTC
    Therefore, you can pass in params for the snapshot date you see on your computer
*/
const snapshotDateToEVMTimestamp = (year, month, day, hours, minutes) => {
    const monthIndex = month - 1; // Date constructor uses monthIndex, not month
    const snapshotDate = new Date(year, monthIndex, day, hours, minutes);
    console.log(snapshotDate); // will show in utc
    const evmTimestamp = Math.floor(snapshotDate.getTime() / 1000); // convert to unix epoch
    console.log(evmTimestamp); // will show in unix epoch (same unit as evm block.timestamp)
}
```

**Review by antoniopgs — APPROVED** [review]
Looks good to me.

File: proposals/10-flex-dapps-proposal.md [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+Pay 30,000 USDC to Flex Dapps and cause 2,000,000 TCR to vest linearly over 3 years with no cliff to Flex Dapps.
+
+## Issue Link
+[#10](https://github.com/tracer-protocol/proposals/issues/10)
+
+## Implementation Discussion
+N/A
+
+## Implementation
+Targets: [`0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48`,`0x707b6be09028e78d2a667db7596b2803c112f9b2`]
```
Comment (raymogg) [inline proposals/10-flex-dapps-proposal.md:11]: @sporejack there is a missing target here - looking below though the implementation looks fine. Just pointing out for clarity and a double check

File: proposals/10-flex-dapps-proposal.md [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+
+## Issue Link
+[#10](https://github.com/tracer-protocol/proposals/issues/10)
+
+## Implementation Discussion
+N/A
+
+## Implementation
+Targets: [`0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48`,`0x9C4A4204B79dd291D6b6571C5BE8BbcD0622F050`, `0x707b6be09028e78d2a667db7596b2803c112f9b2`]
+Data: [`0xa9059cbb000000000000000000000000dd140cb2847f6186a9df3b25ae1e135052b0932100000000000000000000000000000000000000000000000000000006fc23ac00`, `0xa9059cbb000000000000000000000000707b6be09028e78d2a667db7596b2803c112f9b200000000000000000000000000000000000000000000000000000000001e8480`, `0x0d535883000000000000000000000000dd140cb2847f6186a9df3b25ae1e135052b0932100000000000000000000000000000000000000000001a784379d99db420000000000000000000000000000009c4a4204b79dd291d6b6571c5be8bbcd0622f05000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000009c0000000000000000000000000000000000000000000000000000000060a5c2a0`]
```
Comment (raymogg) [inline proposals/10-flex-dapps-proposal.md:12]: The second payload here differs to that produced by the DAOCheck tool, but matches the manually produced code. The difference is the amount. I think the DAOCheck tool has the right amount. Can you confirm this?

Comment (ScaredIbis) [reply proposals/10-flex-dapps-proposal.md:12]: had the same concern, only thing stopping me from approving @sporejack

**Review by raymogg — APPROVED** [review]

**Pull request merged**